src/test/scala/chisel3/tests/AsyncResetRegTest.scala

```diff
@@ -0,0 +1,159 @@
+// See LICENSE for license details.
+
+package chisel3.tests
+
+
+import chisel3._
+import chisel3.core.IntParam
+import chisel3.tester._
+import chisel3.util.{Cat, HasBlackBoxResource}
+import firrtl.ir.{Param, Type}
+import org.scalatest._
+import treadle.executable.{PositiveEdge, Transition}
+import treadle.{ScalaBlackBox, ScalaBlackBoxFactory}
+
+object AsyncResetReg {
+  val bitWidth: Int = 3
+}
+
+class AsyncResetReg(resetValue: Int = 0)
+        extends BlackBox(Map("RESET_VALUE" -> IntParam(resetValue))) with HasBlackBoxResource {
+  val io = IO(new Bundle {
+    val d = Input(Bool())
+    val q = Output(Bool())
+    val en = Input(Bool())
+
+    val clk = Input(Clock())
+    val rst = Input(Bool())
+  })
+
+  setResource("/vsrc/AsyncResetReg.v")
+}
+
+class AsyncResetRegScalaImpl(instanceName: String) extends ScalaBlackBox {
+  override def name: String = "AsyncResetReg"
+
+  var nextValue:    BigInt = BigInt(0)
+  var currentValue: BigInt = BigInt(0)
+  var resetValue:   BigInt = BigInt(0)
+  var enable:       Boolean = false
+
+  def getOutput(inputValues: Seq[BigInt], tpe: Type, outputName: String): BigInt = {
+    currentValue
+  }
+
+  override def inputChanged(name: String, value: BigInt): Unit = {
+    name match {
+      case "d"     => nextValue = value
+      case "en"    => enable = value > BigInt(0)
+      case "rst"   =>
+        if(value > BigInt(0)) {
+          nextValue = resetValue
+          currentValue = nextValue
+        }
+      case _ =>
+    }
+    println(s"next $nextValue cur $currentValue, en $enable")
```

Comment:
Is this println meant to be visible in production?

Reply:
printf removed